repo2docker/contentproviders/git.py

```diff
@@ -44,3 +44,15 @@ def fetch(self, spec, output_dir, yield_output=False):
                                     cwd=output_dir,
                                     capture=yield_output):
                 yield line
+
+        cmd = ['git', 'rev-parse', 'HEAD']
+        sha1 = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd=output_dir)
+        self._sha1 = sha1.stdout.read().decode().strip()
+
+    @property
+    def content_id(self):
+        """A unique ID to represent the version of the content.
+
+        Uses the git commit ID of the repository.
+        """
+        return self._sha1
```

Comment:
Maybe truncate sha1 to 7 or 10 characters here? There's a limit to the length of image names, and adding a full hash could make it a little too easy to bump into that